server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -351,6 +353,26 @@ public TimeValue timeout() {
         }, listener::onFailure);
     }
 
+    public boolean hasOldVersionSnapshots(String repositoryName, RepositoryData repositoryData, @Nullable SnapshotId excluded) {
+        final List<SnapshotId> snapshotIds = List.copyOf(repositoryData.getSnapshotIds());
```

Comment:
why do we need to convert to a list here? Why is Collection not good enough=?

Reply:
My bad, this was a left-over from when I used an API that needed a `List` to get all the snapshot versions before :)